eth/filters/filter.go

```diff
@@ -160,6 +166,9 @@ func (f *Filter) Logs(ctx context.Context) ([]*types.Log, error) {
 	}
 	rest, err := f.unindexedLogs(ctx, end)
 	logs = append(logs, rest...)
+	if f.rangeLimit != 0 {
+		return logs[:f.rangeLimit], err
+	}
```

Comment:
Is this the only place where the rangeLimit is actually enforced/applied? 
I don't fully know how the filter system works, but it looks to me like this only does whatever it did before , and then trim the results on the way out. 

In other words: does this feature really make any difference, from a resource consumption PoV?